system/t12_api/publish.py

```diff
@@ -1036,3 +1036,207 @@ def check(self):
+        self.check_equal(self.post(
+            "/api/repos", json={"Name": repo2_name, "DefaultDistribution": "wheezy"}).status_code, 201)
+
+        d = self.random_name()
+        self.check_equal(self.upload("/api/files/" + d,
+                                     "libboost-program-options-dev_1.49.0.1_i386.deb").status_code, 200)
+
+        self.check_equal(self.post_task("/api/repos/" + repo2_name + "/file/" + d).status_code, 200)
+
+        # publishing under prefix, default distribution
+        prefix = self.random_name()
+        self.post_task(
+            "/api/publish/" + prefix,
+            json={
+                 "SourceKind": "local",
+                 "Sources": [{"Component": "main", "Name": repo1_name}, {"Component": "test", "Name": repo2_name}],
+                 "Signing": DefaultSigningOptions,
+            }
+        )
+
+        # Actual test
+        self.check_task(self.delete("/api/publish/" + prefix + "/wheezy/sources/test"))
+
+        expected = [{"Component": "main", "Name": repo1_name}]
+        response = self.get("/api/publish/" + prefix + "/wheezy/sources")
+        self.check_equal(response.status_code, 200)
+        self.check_equal(expected, response.json())
+
+
+class PublishSourceListAPITestRepo(APITest):
+    """
+    GET /publish/:prefix/:distribution/sources
+    """
+    def check(self):
+        repo1_name = self.random_name()
+        self.check_equal(self.post(
+            "/api/repos", json={"Name": repo1_name, "DefaultDistribution": "wheezy"}).status_code, 201)
+
+        d = self.random_name()
+        self.check_equal(self.upload("/api/files/" + d,
+                                     "libboost-program-options-dev_1.49.0.1_i386.deb", "pyspi_0.6.1-1.3.dsc",
+                                     "pyspi_0.6.1-1.3.diff.gz", "pyspi_0.6.1.orig.tar.gz",
+                                     "pyspi-0.6.1-1.3.stripped.dsc").status_code, 200)
+
+        self.check_equal(self.post_task("/api/repos/" + repo1_name + "/file/" + d).status_code, 200)
+
+        repo2_name = self.random_name()
+        self.check_equal(self.post(
+            "/api/repos", json={"Name": repo2_name, "DefaultDistribution": "wheezy"}).status_code, 201)
+
+        d = self.random_name()
+        self.check_equal(self.upload("/api/files/" + d,
+                                     "libboost-program-options-dev_1.49.0.1_i386.deb").status_code, 200)
+
+        self.check_equal(self.post_task("/api/repos/" + repo2_name + "/file/" + d).status_code, 200)
+
+        # publishing under prefix, default distribution
+        prefix = self.random_name()
+        self.post_task(
+            "/api/publish/" + prefix,
+            json={
+                 "SourceKind": "local",
+                 "Sources": [{"Component": "main", "Name": repo1_name}],
+                 "Signing": DefaultSigningOptions,
+            }
+        )
+
+        # Actual test
+        self.check_equal(
+            self.post(
```

Comment:
use post_task here ? then 201 and a task is returned...
otherwise return code is 200 and request is executed synchronously (api returns after completion)